### DISCUSSION TITLE
more than 15000 Hz

### DISCUSSION THREAD

**Discussion by malozaibi** [ggerganov/ggwave#47] [Q&A] [answered]
I dug in the library and found how to increase the freqStart and volume.

I wanted it to be inaudible at all. (the default ultrasonic mode (15000) can be heard a bit)

so my question is will the phone speaker be damaged if I increased it to 17000 Hz (freqStart)?
and the volume if I made it 45, will it damage it too?

Thanks

**Comment by ggerganov** [comment, marked as answer]
Hi, I believe it should be OK to increase the frequency and volume as you suggested.

However, please note that I cannot give you any kind of guarantee that the speaker will not get damaged in the long run.
It depends on the specification of your hardware and if it can handle these type of frequencies.

**Reply by malozaibi** [reply]
Thank you so much for quick answer,

one more question,
I want to send the sound message **as far as possible** in ultrasonic,
what configurations do you recommend without hurting the speaker?

- the frequency
- the volume in the library
- the mobile phone ringtone volume

thanks in advance

**Reply by ggerganov** [reply]
## Frequency

If you are using `bytesPerTx = 3`, I would recommend to not go above `freqStart = 330, (i.e. ~15500 Hz)` because the width of the channel is `4500 Hz`, so the highest frequency will become `15500 + 4500 = 20000 Hz` which as far as I know is the upper limit for most speakers/microphones. As far as I understand, going further would not necessarily damage the speaker - it's just the speaker would not be able to produce the sound. It's best to use some sort of real-time [spectrogram](https://spectrogram.ggerganov.com) and investigate the frequency range of your hardware.

If you decrease `bytesPerTx = 2` , then the channel width becomes `3000 Hz` and you can go up to `freqStart = 362, (i.e. ~17000 Hz)` without loosing any part of the channel.

If you decrease even further to `bytesPerTx = 1`, then the channel width becomes `1500 Hz` and you can go up to `freqStart = 394, (i.e. ~18500 Hz)` without loosing information.

Of course, decreasing the `bytesPerTx` will make the transmission slower.

## Volume

I don't think there is any reason to increase the volume too much for ultrasound messages, since `ggwave` uses a type of FSK encoding.  At these frequencies the background noise is typically much lower, so it's easier to detect the signal even at lower volume.
I would recommend to keep the volume in the library at 25 for example, and on the mobile - anywhere between 25-75% should be OK.

Again, these are just recommendations based on my own experience and hardware, so there might be better parameters for your specific case.

If your use case is to transmit messages with known length, then I would also recommend to use the "fixed length" option of `ggwave`:

https://github.com/ggerganov/ggwave/blob/master/include/ggwave/ggwave.h#L65-L70

**Reply by malozaibi** [reply]
Extreme appreciation to you,

This helped me a lot. I will use fixed length and lower `byptePerTX` and volume as you described and test these options (I will send small data only).

I am naive so I taught increasing volume will make the signal reach longer distance. I want the signal to travel to the end of the classroom (big classroom) as I will use it as an attendance program for our graduation project.

If there is any advice to help me in my use case I would be grateful,

**Reply by danielcamposramos** [reply]
What if you don't use speaker and microphone, but a, let's say cat5  pair of wires, could it work?

**Comment by marcos-martins** [comment]
Hello, @malozaibi and @ggerganov Maybe this is a silly question, but I couldn't find how to increase the freqStart. I'm willing to test some frequencies between 17000 Hz and 18500 Hz. 
Can you help me, I really appreciate it.

Thanks
